shaarli/python-shaarli-client
# Add Tox and Travis CI configuration

Tox is a tool to automate the creation/renewal of Python test environments
(virtualenvs) and running commands within them:
- static analysis
- test harnesses
- documentation generation
- packaging
- ...

Added:
- Travis CI configuration
- Tox configuration
- `docs` test environment to:
  - check Sphinx docs build without errors and warnings
  - check rST pages are well-formatted

See:
- https://tox.readthedocs.io/en/latest/
- https://tox.readthedocs.io/en/latest/config.html
- https://tox.readthedocs.io/en/latest/config.html#confval-basepython=NAME-OR-PATH
- https://virtualenv.pypa.io/en/stable/
- https://virtualenv.pypa.io/en/stable/userguide/#usage

Notes:
- the `docs` environment will be based on the system's default python3 interpreter
- tox creates a virtualenv for each testenv, located under `.tox`
- these virtualenvs can also be activated for debugging purposes
  (e.g. check dependency combination/resolution):
  `$ source .tox/ENV1/bin/activate`

Running Tox:
- install tox system-wide or within a virtualenv
- to create testenvs if needed and run all tests:
  `$ tox`
- to recreate testenvs and run all tests:
  `$ tox -r`
- to recreate and run tests on (a) given testenv(s):
  `$ tox -r -e ENV1 -e ENV2`
